sdccc/src/test/java/com/draeger/medical/sdccc/manipulation/precondition/impl/ManipulationPreconditionsTest.java

```diff
@@ -331,10 +653,59 @@ void testAssociateNewLocationForHandleSameStateTwice() {
                 .thenReturn(Optional.of(LOCATION_CONTEXT_STATE_HANDLE))
                 .thenReturn(Optional.empty());
 
+        final MdibAccessObservable mockMdibAccessObservable = mock(MdibAccessObservable.class);
+        when(mockTestClient.getSdcRemoteDevice().getMdibAccessObservable()).thenReturn(mockMdibAccessObservable);
+        final AtomicReference<MdibAccessObserver> observer = new AtomicReference<>();
+        final Object signal = new Object();
+        doAnswer((arguments) -> {
+                    synchronized (signal) {
+                        observer.set(arguments.getArgument(0));
+                        signal.notifyAll();
+                    }
+                    return null;
+                })
+                .when(mockMdibAccessObservable)
+                .registerObserver(any());
+
+        final Thread t = new Thread(() -> {
+            ManipulationPreconditions.AssociateLocationsManipulation.EpisodicContextReportStateHandleObserver
+                    mdibAccessObserver;
+            for (int i = 0; i < 2; i++) {
+                mdibAccessObserver =
+                        (ManipulationPreconditions.AssociateLocationsManipulation
+                                        .EpisodicContextReportStateHandleObserver)
+                                observer.get();
+                synchronized (signal) {
+                    while (mdibAccessObserver == null) {
+                        try {
+                            signal.wait(300);
+                        } catch (InterruptedException e) {
```

Comment:
do not catch it here if you are not reacting to it

also you are not even interrupting the thread from another thread, thus there is no good way of reacting to it and it should just be escalated

Reply:
same as above.

Reply:
Improved just like above.